_posts/2025-01-31-wildfly-domain-usage.adoc

```diff
@@ -0,0 +1,401 @@
+Added user 'admin' to file '/wildfly-35.0.0.Final/domain/configuration/mgmt-users.properties'
+....
+
+From the above command output, we can see the `admin` user is added and its password is `123`. This added user will be used on the Secondary Host Controller to connect to the Primary Host controller.
+
+Please note that we didn’t generate a secret value related to the generated `admin` user, which is already deprecated. So this kind of configuration on the Secondary Host Controller is deprecated:
+
+[source,xml]
+----
+<server-identities>
+  <secret value="..." />
+</server-identities>
+----
+
+As the Primary Host Controller is started on Machine A, now we can check the WildFly server configuration on Machine B. As planned, the Host Controller is named `secondary`, and it will be connected to the Primary Host Controller on Machine A, which acts as the domain controller.
+
+The Secondary Host Controller on Machine B will use the `host-secondary.xml` as its configuration file. We need to make some modifications to this file. First I need to add the `name` property to the `host` section:
+
+[source,xml]
+----
+<host xmlns="urn:jboss:domain:community:20.0" name="secondary">
+----
+
+Giving the host a name can help to analyze the server log output later. Then I need to add another configuration to the `domain-controller`:
+
+[source,xml]
+----
+<domain-controller>
+    <remote authentication-context="myCtx">
+        <discovery-options>
+            <static-discovery name="primary" protocol="${jboss.domain.primary.protocol:remote+http}"
+                              host="${jboss.domain.primary.address}" port="${jboss.domain.primary.port:9990}"/>
+        </discovery-options>
+    </remote>
+</domain-controller>
+----
+
+As the configuration is shown above, I added a `remote` configuration into the `domain-controller` section, and defined the `authentication-context` to be used as `myCtx`. This configuration is different from the legacy authentication. So I need to define this `authentication-context` in the `elytron` subsystem. Here are the details:
+
+[source,xml]
+----
+<profile>
+    <subsystem xmlns="urn:jboss:domain:core-management:1.0"/>
+    <subsystem xmlns="urn:wildfly:elytron:community:18.0" final-providers="combined-providers"
+               disallowed-providers="OracleUcrypto" register-jaspi-factory="false">
+        <authentication-client>
+            <authentication-configuration sasl-mechanism-selector="DIGEST-MD5" name="myConfig"
+                                          authentication-name="admin" realm="ManagementRealm">
+                <credential-reference clear-text="123"/>
+            </authentication-configuration>
+            <authentication-context name="myCtx">
+                <match-rule match-host="${jboss.domain.primary.address}" authentication-configuration="myConfig"/>
+            </authentication-context>
+        </authentication-client>
+        ...
+    </subsystem>
+    ...
+</profile>
```

Comment:
We should do this with CLI IMO. Probably before the step above.

```
/host=stalefish/subsystem=elytron/authentication-configuration=myConfig:add(sasl-mechanism-selector=DIGEST-MD5, authentication-name=admin, realm=ManagementRealm)
/host=stalefish/subsystem=elytron/authentication-context=myCtx:add(match-rules=[{match-host="${jboss.domain.primary.address}", authentication-configuration=myConfig}])
```

Reply:
@jamezp Is there way to configure the password in the CLI command?

Reply:
I tried out the command to add the config with credential:

```
[domain@embedded /] /host=secondary/subsystem=elytron/authentication-configuration=myConfig:add(sasl-mechanism-selector=DIGEST-MD5, authentication-name=admin, realm=ManagementRealm, credential-reference={clear-text="123"})
{"outcome" => "success"}
```

Generated config:

```xml
<authentication-client>
                <authentication-configuration name="myConfig" authentication-name="admin" realm="ManagementRealm" sasl-mechanism-selector="DIGEST-MD5">
                    <credential-reference clear-text="123"/>
                </authentication-configuration>
            </authentication-client>
```

Reply:
I have updated the text accordingly. Thanks for teaching James!